File: dotnet/src/Grafana.Agento11y/SdkVersion.cs
namespace Grafana.Agento11y;

/// <summary>
/// SDK version and User-Agent product token.
/// </summary>
/// <remarks>
/// <see cref="Version"/> is stamped into the default generation-export User-Agent
/// (see <see cref="UserAgent"/>). Keep in sync with the package version on release.
/// </remarks>
public static class SdkVersion
{
    /// <summary>Released version of the Sigil .NET SDK.</summary>
    public const string Version = "0.1.0";

    private const string UserAgentProduct = "agento11y-sdk-dotnet";

    /// <summary>
    /// Returns the SDK's default generation-export User-Agent product token,
    /// <c>agento11y-sdk-dotnet/&lt;Version&gt;</c>.
    /// </summary>
    public static string UserAgent() => UserAgentProduct + "/" + Version;
}

File: java/core/src/main/java/com/grafana/agento11y/sdk/SdkVersion.java
package com.grafana.agento11y.sdk;

/**
 * SDK version and User-Agent product token.
 *
 * <p>{@link #VERSION} is stamped into the default generation-export User-Agent (see {@link
 * #userAgent()}). Keep in sync with the gradle {@code version} on release.
 */
public final class SdkVersion {
    /** Released version of the Sigil Java SDK. */
    public static final String VERSION = "0.2.0";

    private static final String USER_AGENT_PRODUCT = "agento11y-sdk-java";

    private SdkVersion() {}

    /**
     * Returns the SDK's default generation-export User-Agent product token, {@code
     * agento11y-sdk-java/<VERSION>}.
     */
    public static String userAgent() {
        return USER_AGENT_PRODUCT + "/" + VERSION;
    }
}
